src/components/select/select-list/select-list.component.js

```diff
@@ -34,9 +34,33 @@ const SelectList = React.forwardRef(
     const listRef = useRef();
     const listActionButtonRef = useRef();
 
+    const getChildOptions = (allChildren) => {
+      return React.Children.toArray(allChildren).filter(
+        (child) => !child.props.label
+      );
+    };
+
+    const getScrollIndex = useCallback(
+      (childOptionIndex) => {
+        const currentSelectedOption = getChildOptions(children)[
+          childOptionIndex
+        ];
+        let currentOptionValue;
+
+        if (currentSelectedOption) {
+          currentOptionValue = currentSelectedOption.props.value;
+        }
+
+        return React.Children.toArray(children).findIndex(
+          (child) => child.props.value === currentOptionValue
```

Comment:
Could `child.props.value` not be an object?

Reply:
Although I see below there are other places where it's being treated as if it's only a string so maybe the work to update for objects as well should be done separately

Reply:
Yeah like you say, it's already being treated as a string, I think it's on Kuba's radar to make all this stuff work with it being an object as well, so this can be changed at the same time as everything else

Reply:
Yeah makes sense